### PR TITLE
added function for displaying boundaries on map & resulting HTML page

### PR DESCRIPTION
**Pull request by tomw66**

### CODE REVIEW

**Comment by lucascatherine** [comment]
you will also need to edit test_generate_map_based_visualisations.py to add tests for the boundary map generation. You can follow the format of the other tests here.

File: tests/unit/visualise/test_generate_map_based_visualisations.py [hunk 1]
```diff
@@ -76,3 +86,18 @@ def test_aircraft_track_map_has_children(aircraft_track):
     # Again, this input file is fixed and static so the output should always
     # produce three children.
     assert len(aircraft_track._children) == 3
+
+
+# Tests for get_boundaries:
+def test_boundaries_map_is_Map(boundaries):
+    """Test that boundary data has been successfully converted to a folium
+    Map object."""
+    assert boundaries._name == 'Map'
```
Comment (crbunney) [inline tests/unit/visualise/test_generate_map_based_visualisations.py:95]: A better way to test this is to use `isinstance`:
```suggestion
    assert isinstance(boundaries, folium.Map)
```

File: src/clean_air/visualise/generate_map_based_visualisations.py [hunk 2]
```diff
@@ -98,3 +96,23 @@ def get_aircraft_track_map(aircraft_track_coords, output_path) -> map:
     m5.save(output_path)
 
     return m5
+
+def get_boundaries(boundary_data, output_path) -> map:
```
Comment (crbunney) [inline src/clean_air/visualise/generate_map_based_visualisations.py:100]: This return type doesn't look valid to me because I can't find what `map` references. It looks like it maybe should be `folium.Map`
```suggestion
def get_boundaries(boundary_data, output_path) -> folium.Map:
```

**Review by crbunney — COMMENTED** [review]
Looks good, I made a few minor comments that it'd be good if you could check out

File: tests/unit/visualise/test_generate_map_based_visualisations.py [hunk 3]
```diff
@@ -24,21 +24,31 @@ def aircraft_filepath(sampledir):
...
 def aircraft_track(aircraft_filepath, tmp_output_path):
     save_path = os.path.join(tmp_output_path, "AircraftTrack.html")
     aircraft_track = make_maps.get_aircraft_track_map(aircraft_filepath,
                                                       save_path)
     return aircraft_track
 
+@pytest.fixture()
+def boundaries(boundaries_filepath, tmp_output_path):
+    save_path = os.path.join(tmp_output_path, "boundaries.html")
+    boundaries = make_maps.get_boundaries(boundaries_filepath,
```
[the remark this answers is not in the document]
Comment (tomw66) [reply tests/unit/visualise/test_generate_map_based_visualisations.py:49]: Done for all 3 functions (though I've left the filepath definitions as is?

File: tests/unit/visualise/test_generate_map_based_visualisations.py (hunk 1 above)
Comment (tomw66) [reply tests/unit/visualise/test_generate_map_based_visualisations.py:95]: Done for all 3 functions

File: src/clean_air/visualise/generate_map_based_visualisations.py (hunk 2 above)
Comment (tomw66) [reply src/clean_air/visualise/generate_map_based_visualisations.py:100]: Done for all 3 functions

File: tests/unit/visualise/test_generate_map_based_visualisations.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (crbunney) [reply tests/unit/visualise/test_generate_map_based_visualisations.py]: yea, that makes sense. As input to the "structure under test" the filepath is something that falls under test setup

[the remark this answers is not in the document]
Comment (tomw66) [reply tests/unit/visualise/test_generate_map_based_visualisations.py]: finally done, does this make more sense?

File: tests/unit/visualise/test_generate_map_based_visualisations.py [hunk 4]
```diff
@@ -18,6 +19,12 @@ def AURN_filepath(sampledir):
     return AURN_filepath
 
 
+@pytest.fixture()
+def AURN_savepath(tmp_output_path):
+    AURN_savepath = os.path.join(tmp_output_path, "AURN.html")
+    return AURN_savepath
```
Comment (crbunney) [inline tests/unit/visualise/test_generate_map_based_visualisations.py:25]: your variable has the same name as the function its in. I think this works ok because the variable only (shadows)[https://en.wikipedia.org/wiki/Variable_shadowing] the function name within the scope of that function, but it's best to give it a different name

File: tests/unit/visualise/test_generate_map_based_visualisations.py [hunk 5]
```diff
@@ -18,6 +19,12 @@ def AURN_filepath(sampledir):
     return AURN_filepath
 
 
+@pytest.fixture()
+def AURN_savepath(tmp_output_path):
```
Comment (crbunney) [inline tests/unit/visualise/test_generate_map_based_visualisations.py:23]: Whilst we're tweaking this, could you change the names of these `AURN_*` functions to be lower case, like this
```suggestion
def aurn_savepath(tmp_output_path):
```
As that's the convention for function names (Names starting with uppercase are the convention for classes)

**Review by crbunney — COMMENTED** [review]
That's looking good, we're almost over the line. There's just a couple naming things to straighten out (see the comments)

File: tests/unit/visualise/test_generate_map_based_visualisations.py (hunk 4 above)
Comment (tomw66) [reply tests/unit/visualise/test_generate_map_based_visualisations.py]: Done!

File: tests/unit/visualise/test_generate_map_based_visualisations.py [hunk 6]
```diff
@@ -1,3 +1,4 @@
+from importlib.resources import path
```
Comment (crbunney) [inline tests/unit/visualise/test_generate_map_based_visualisations.py:1]: I can't see where this is used, is it needed? (it's also shadowed by local variables with the same name)

Comment (tomw66) [reply tests/unit/visualise/test_generate_map_based_visualisations.py]: Removing it hasn't caused any issues, so I guess not!

**Review by crbunney — APPROVED** [review]
